### PR TITLE
added orderByTranslation scope

### PR DESCRIPTION
**Pull request by justijndepover**
I made a new scope to order results by translation fields.

```
Countries::orderByTranslation('name')->get();
```
will order the results based on the `name` attribute inside the translation table. 

```
Countries::orderByTranslation('name', 'desc')->get();
```
will do the same but in descending order

### CODE REVIEW

**Comment by Gummibeer** [comment]
StyleCI will be fixed in #454

File: src/Translatable/Translatable.php [hunk 1]
```diff
@@ -674,6 +674,30 @@ public function scopeOrWhereTranslationLike(Builder $query, $key, $value, $local
...
+     * @return \Illuminate\Database\Eloquent\Builder|static
+     */
+    public function scopeOrderByTranslation(Builder $query, $key, $sortmethod = 'asc')
+    {
+        $translationTable = $this->getTranslationsTable();
+        $localeKey = $this->getLocaleKey();
+        $table = $this->getTable();
+        $keyName = $this->getKeyName();
+
+        return $query->join($translationTable, function($join) use ($translationTable, $localeKey, $table, $keyName) {
```
Comment (Gummibeer) [inline src/Translatable/Translatable.php]: `$join` is also instance of `Builder` and could be type-hinted.

File: src/Translatable/Translatable.php [hunk 2]
```diff
@@ -674,6 +674,30 @@ public function scopeOrWhereTranslationLike(Builder $query, $key, $value, $local
...
+            $join->on($translationTable.'.'.$this->getRelationKey(), '=', $table.'.'.$keyName)
```
Comment (Gummibeer) [inline src/Translatable/Translatable.php]: I don't like the indent and linebreaks in this whole block.

I would like it more:
```php
$query
  ->join(function() {
    $join
      ->on()
      ->where();
  })
  ->orderBy()
  ->select()
  ->with();
```

But this is just me and won't be a blocker.

**Comment by Gummibeer** [comment]
@dimsav author seems inactive here - I've fixed my two change requests. But StyleCI will fail until master is merged. How should we proceed here?

**Comment by dimsav** [comment]
@Gummibeer cool! We still need to update the docs before merging. That's fine for styleci, we can fix later on master.

**Comment by Gummibeer** [comment]
**blocked by:** #486

**Comment by justijndepover** [comment]
Sorry for inactivity. Do you guys still need something from me?

**Comment by Gummibeer** [comment]
@justijndepover it's your branch so if you could merge the current master into this one would be nice. Just to run styleci on this PR.

After this it's still blocked to get merged atm - but this is the only thing missing by you. 😉

**Comment by justijndepover** [comment]
done

**Comment by Gummibeer** [comment]
StyleCI is still failing but the issue is already fixed in #468 - to prevent conflicts I won't fix it here.

**Comment by vlados** [comment]
@justijndepover are you going to fix the problems with the merge?

**Comment by Gummibeer** [comment]
@vlados do you mean the missing docs? Nothing more is missing. And in case we can add it after this PR.

**Comment by justijndepover** [comment]
I believe the only thing left to do is to merge the pull request?

**Comment by vlados** [comment]
I meant can it be merged?

**Pull request merged**